### DISCUSSION TITLE
Zugriff auf OpenDTU per Apache2 und Reverse-Proxy

### DISCUSSION THREAD

**Discussion by jimjackii** [tbnobody/OpenDTU#1206] [General]
Hallo in die Runde,

ich habe zu dem Thema nichts weiter gefunden oder vielleicht übersehen. Seht es mir nach. ;-)

Ich würde gerne auf OpenDTU per Apache2 mit Reverse-Proxy zugreifen.

Hier meine 2 Varianten der Apache2-Konfiguration.

1.
ProxyPass /bkw/ http://192.168.xx.yy/
ProxyPassReverse /bkw/ http://192.168.xx.yy/

2.
<Location /bkw/>
  ProxyPass http://192.168.xx.yy/
  ProxyPassReverse http://192.168.xx.yy/
  RewriteEngine on
  RewriteCond %{HTTP:CONNECTION} Upgrade$ [NC]
</Location>

Leider klappt es mit keine der Variante, die OpenDTU-Seite zu sehen. Es wird nur die Statuszeile im Webbrowser angezeigt.
Daten bekommen ich aber leider nicht zu sehen.

Danke.

VG Steffen

**Comment by k1llerk3ks** [comment]
Hey, ich habe leider keine Erfahrung mit Apache2. Da ich auch kein Beispiel für `nginx` gefunden habe, habe ich kurzerhand eins selbst geschrieben. Als besonders schwer hab ichs nicht empfunden, aber ob alle Parameter so notwendig sind, hab ich nicht überprüft. Funktioniert allerdings bestens soweit. PS: Wofür brauchst du ein ProxyPassReverse?

```
server {

    listen      80;
    listen      [::]:80;

    server_name "dtu.domain.com";

    return 301 https://dtu.domain.com$request_uri;

}

server {

    server_name "dtu.domain.com";

    listen [::]:443 ssl http2; # managed by Certbot
    listen 443 ssl http2; # managed by Certbot

    include /etc/nginx/includes/ssl.conf;

    ssl_certificate /etc/letsencrypt/live/dtu.domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/dtu.domain.com/privkey.pem; # managed by Certbot

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $http_host;
        proxy_pass http://<DTU-IP-replaceme>;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_next_upstream error timeout http_502 http_503 http_504;
    }
}
```

**Reply by jimjackii** [reply]
Danke, für die Config.

Ich schaue mir das an und versuche es für den Apache2 zu "übersetzen". ;-).

**Reply by Matthy1810** [reply]
Bei mir klappt damit das Einloggen nicht. Bekomme ständig Error 401 beim Zugriff auf /api. Ich benutze ebenfalls nginx. @k1llerk3ks, klappt das bei dir?

**Reply by k1llerk3ks** [reply]
@Matthy1810 Ich kann das nicht bestätigen, bei mir klappt die Config einwandfrei, gerade nochmal getestet. Ich schätze dementsprechend, der Fehler liegt irgendwo anders. Mein Nginx läuft einfach auf nem Raspberry, aber ist ja eigentlich wuppe, was für ein OS das ist.

**Reply by Matthy1810** [reply]
Danke, @k1llerk3ks. Ja, es lag an der Authentifizierung, die ich per auth_basic Option noch hizugefügt hatte. Wenn ich die entferne, geht es. Ich kann sie aber leider nicht weglassen, daher werde ich nun schauen ob ich eine Lösung mit Basic-Auth finde.

**Reply by Matthy1810** [reply]
OK, habe es gelöst: OpenDTU benutzt selbst auch den Basic-Auth-Header. Wenn ich in nginx den Header mit den passenden Credentials mittels "proxy_set_header Authorization" setze, klappt es und ich bin dann nach dem Basic-Auth-Login direkt auch in OpenDTU eingeloggt.
